website/static/js/filepage/index.js

```diff
@@ -215,7 +216,58 @@ var FileViewPage = {
                 m('button.btn.btn-sm.btn-danger.file-delete', {onclick: $(document).trigger.bind($(document), 'fileviewpage:delete')}, 'Delete')
             ]) : '',
             m('.btn-group.m-t-xs', [
-                m('button.btn.btn-sm.btn-primary.file-download', {onclick: $(document).trigger.bind($(document), 'fileviewpage:download')}, 'Download')
+                m('#sharebutton.btn.btn-sm.btn-primary.file-share', {onclick: function () {
```

Comment:
There's a lot going on here; could you separate the share button into its own component, so this would become

```js
m('.btn-group.m-t-xs', [
    m.component(ShareButton)
])
```

Breaking this out into its own component will help with readability and reusability